components/scream/src/dynamics/homme/physics_dynamics_remapper.hpp

```diff
@@ -589,396 +994,493 @@ setup_boundary_exchange () {
+        auto dyn = Unmanaged<KokkosTypes<DefaultDevice>::view<ScalarT**[NP][NP]>> (reinterpret_cast<ScalarT*>(dyn_ptrs(i).get()),
+                                                                                   dyn_dims(i,0),
+                                                                                   dyn_dims(i,1));
+
+        const auto tr = Kokkos::TeamThreadRange(team, num_cols*dim);
+        const auto f = [&] (const int idx) {
+          const int icol = idx/dim;
+          const int idim = idx%dim;
 
           const auto& elgp = Kokkos::subview(lid2elgp,p2d(icol),Kokkos::ALL());
           dyn(elgp[0],idim,elgp[1],elgp[2]) = phys(icol,idim);
-        });
+        };
+        Kokkos::parallel_for(tr, f);
       }
       break;
     }
-    case LayoutType::Tensor2D:
+    case 2:
     {
-      auto phys = phys_field.template get_reshaped_view<Real***>();
-      auto dyn  = dyn_field.template get_reshaped_view<Real***[NP][NP]>();
-      const int dim1 = phys_dims[1];
-      const int dim2 = phys_dims[2];
-      Kokkos::pair<int,int> ordering;
-      if (phys_layout.tag(1)==dyn_field.get_header().get_identifier().get_layout().tag(1)) {
-        ordering.first=0;
-        ordering.second=1;
-      } else {
-        ordering.first=1;
-        ordering.second=0;
-      }
-      Kokkos::parallel_for(RangePolicy(0,num_cols*dim1*dim2),
-                           KOKKOS_LAMBDA(const int idx) {
-        const int icol = idx / (dim1*dim2);
-        const int dims [2] = { (idx/dim2)%dim1 , idx%dim2 };
+      auto phys = Unmanaged<KokkosTypes<DefaultDevice>::view<ScalarT***>> (reinterpret_cast<ScalarT*>(phys_ptrs(i).get()),
+                                                                           phys_dims(i,0),
+                                                                           phys_dims(i,1),
+                                                                           phys_dims(i,2));
+      auto dyn = Unmanaged<KokkosTypes<DefaultDevice>::view<ScalarT***[NP][NP]>> (reinterpret_cast<ScalarT*>(dyn_ptrs(i).get()),
+                                                                                  dyn_dims(i,0),
+                                                                                  dyn_dims(i,1),
+                                                                                  dyn_dims(i,2));
+
+      const int dim1 = phys_dims(i, 1);
+      const int dim2 = phys_dims(i, 2);
+
+      const auto tr = Kokkos::TeamThreadRange(team, num_cols*dim1*dim2);
+      const auto f = [&] (const int idx) {
+        const int icol = idx/(dim1*dim2);
+        const int dims[2] = { (idx/dim2)%dim1 , idx%dim2 };
 
         const auto& elgp = Kokkos::subview(lid2elgp,p2d(icol),Kokkos::ALL());
-        dyn(elgp[0],dims[ordering.first],dims[ordering.second],elgp[1],elgp[2]) = phys(icol,dims[0],dims[1]);
-      });
+
+        dyn(elgp[0],dims[ordering(i,0)],dims[ordering(i,1)],elgp[1],elgp[2]) = phys(icol,dims[0],dims[1]);
```

Comment:
I can't think of any Tensor quantity to remap, but I'm ok keeping the capability. However, it might trim down this class if we _enforced_ the two dimensions to appear in the _same_ order. What do you think? The hardcore solution would be to completely remove support for Tensor2D and Tensor3D. Which I _think_ PD remapping doesn't really _need_.

Reply:
Either way. If it's not needed then I say just remove.. But I'm not sure what all fields would need to be remapped in the future? 

If we keep it should we just `assert(phys_layout.tag(1) == dyn_layout.tag(1))`?

Reply:
I think we can always add this code back if need be. We can always go back on the PR and see the code that got deleted. My guess, however, is that we won't need this, ever. So yeah, I am also in favor of purging it, making the class maintenance easier.